frontend/src/utils/comparison/pending.ts

```diff
@@ -0,0 +1,165 @@
+import { PollCriteria } from 'src/services/openapi';
+import { CriteriaValuesType } from 'src/utils/types';
+
+// Name of the key used in the browser's local storage.
+const PENDING_NS = 'pendingComparisons';
+
+const initPending = () => '{}';
+
+const makePendingKey = (
+  poll: string,
+  uidA: string,
+  uidB: string,
+  criterion: string
+) => {
+  return `${poll}/${uidA}/${uidB}/${criterion}`;
+};
+
+const keyIsInvalid = (
+  poll: string,
+  uidA: string,
+  uidB: string,
+  criterion: string | null
+) => {
+  return [poll, uidA, uidB, criterion].includes('');
+};
+
+/**
+ * Save a criterion rating for a couple of entities in a given poll.
+ *
+ * The rating is saved in the browser's local storage, and can be retrieved
+ * with `getPendingRating`.
+ *
+ * The ratings are saved with the following structure:
+ *
+ *  {
+ *     `${pollName}/${uidA}/${uidB}`: ${rating},
+ *  }
+ *
+ * @param poll The name of the poll.
+ * @param uidA UID of the entity A.
+ * @param uidB UID of the entity B.
+ * @param criterion Name of the criterion ralated to the rating.
+ * @param rating A score given by a user.
+ */
+export const setPendingRating = (
+  poll: string,
+  uidA: string,
+  uidB: string,
+  criterion: string,
+  rating: number
+) => {
+  let pending = localStorage.getItem(PENDING_NS);
+
+  if (pending == null) {
+    pending = initPending();
+  }
+
+  if (keyIsInvalid(poll, uidA, uidB, criterion)) {
+    return null;
+  }
+
+  const pendingJSON = JSON.parse(pending);
+  pendingJSON[makePendingKey(poll, uidA, uidB, criterion)] = rating;
+
+  localStorage.setItem(PENDING_NS, JSON.stringify(pendingJSON));
+};
+
+/**
+ * Retrieve a previously saved criterion rating for a couple of entities in a
+ * given poll.
+ *
+ * If `pop` is true, the rating is deleted from the browser's local storage.
+ */
+export const getPendingRating = (
+  poll: string,
+  uidA: string,
+  uidB: string,
+  criterion: string,
+  pop?: boolean
+): number | null => {
+  const pending = localStorage.getItem(PENDING_NS);
+
+  if (pending == null) {
+    return null;
+  }
+
+  if (keyIsInvalid(poll, uidA, uidB, criterion)) {
+    return null;
+  }
+
+  const pendingJSON = JSON.parse(pending);
+  const pendingKey = makePendingKey(poll, uidA, uidB, criterion);
+
+  const rating = pendingJSON[pendingKey];
+
+  if (pop) {
+    delete pendingJSON[pendingKey];
+    localStorage.setItem(PENDING_NS, JSON.stringify(pendingJSON));
+  }
+
+  return rating ? rating : null;
```

Comment:
It's a bit tricky but yes, 0 means at the same time a pending rating of 0 or no pending rating.

I'm going to use the `??` operator here, to make the function actually return the saved rating, even if it's 0. I'm not sure the current implementation will cause a problem, but i think it's more elegant to return 0 when 0 is saved, instead of returning null.